tools/validation_files.py

```diff
@@ -1,46 +1,50 @@
     #else: #json file
         #TODO call to function that return dict from json file
-    samples_from_barcode2sample=sorted(load_table_to_dict(barcode2samples_path,'Barcode {} belongs to more than one sample!!').values())
+    samples_from_barcode2sample = sorted(load_table_to_dict(barcode2samples_path,'Barcode {} belongs to more than one sample!!').values())
     return samples_from_sample2bc==samples_from_barcode2sample
 
 def is_valid_structure_file(file_path):
-    #verift the structure of the file, should look like: word\tword\n
-    #word can conation: number, big or little letter,_
+    # verify the structure of the file -  should look like: word\tword\n
+    # word can conation: number, big or little letter and _
     with open(file_path, "rb") as f:
-        b=f.read(1)
-        find_tab=False
-        count_tab=0
+        b = f.read(1)
+        find_tab = False
+        count_tab = 0
         while b:
-            num=int.from_bytes(b, byteorder=sys.byteorder)
+            num = int.from_bytes(b, byteorder=sys.byteorder)
             if num==9: #is a TAB
-                count_tab+=1
-                find_tab=True
+                count_tab += 1
+                find_tab = True
             elif num==10 and find_tab and count_tab==1: #10 is new line
-                find_tab=False
-                count_tab=0
+                find_tab = False
+                count_tab = 0
             elif num<48 or (num>57 and num<65) or (num>90 and num<95) or (num>95 and num<97) or (num>122):
                 #48-57 number, 65-90 Big letter, 97-122 small letter, 95 is _
                 return False
-            b=f.read(1)
+            b = f.read(1)
     return True        
 
 def is_validation_files(samples2bc_path, barcode2samples_path):
-    samples2bc_valid=True
-    filename, file_extension = os.path.splitext(samples2bc_path)
-    if file_extension=='txt':
-        samples2bc_valid = is_same_samples(samples2bc_path)
-    barcode2samples_valid = is_same_samples(barcode2samples_path)
-    if samples2bc_valid and barcode2samples_valid:
-        return is_same_samples(samples2bc_path, barcode2samples_path)
+    samples2bc_valid = False
+    if samples2bc_path:
+        filename, file_extension = os.path.splitext(samples2bc_path)
+        samples2bc_valid = is_valid_structure_file(samples2bc_path) if file_extension=='.txt' else True
+        print(f'samples2bc_valid: {samples2bc_valid}')
+    barcode2samples_valid = is_valid_structure_file(barcode2samples_path) if barcode2samples_path else False
+    print(f'barcode2samples_valid: {barcode2samples_valid} ')
+    if samples2bc_path and barcode2samples_path:
+        if samples2bc_valid and barcode2samples_valid:
+            return is_same_samples(samples2bc_path, barcode2samples_path)
+        else:
+             False
     else:
-        return False
\ No newline at end of file
+        return samples2bc_valid or barcode2samples_valid
```

Comment:
shouldn't be `and` instead of `or`?

Reply:
no, it is a case when there is only one file as input so it should be at least one True

Reply:
But if we have both file and one if invalid the function will return that all is valid...

Reply:
Code should be clearer/simpler to follow.
Prefer early return, i.e., in `if samples2bc_path:` you check about `barcode2samples_path` value and if empty return and vice-versa.

Reply:
but most of the time the input for the function is only one file and not two of them so if there is no file I can't do return.